### PR TITLE
814 UI for timeline block

### LINKED ISSUE #814: [Solution] UI for timeline menu and timeline block

### PR DESCRIPTION
**Pull request by rgravitvl**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #814 

Test URLs:
- Before: https://main--danaher-ls-aem--hlxsites.hlx.page/us/en/solutions/mabs/cell-line-development/process-steps
- After: https://814-ui-for-timeline-block--danaher-ls-aem--hlxsites.hlx.page/us/en/solutions/mabs/cell-line-development/process-steps

### CODE REVIEW

File: blocks/timeline/timeline.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    const svg = title.querySelector('svg');
+    svg?.remove();
+  });
+
+  const title = clickedMenu.querySelector('div > h2');
+  clickedMenu.classList.add('border-danaherlightblue-500');
+  title.innerHTML += `
+        <svg data-v-e019efe8="" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="currentColor" aria-hidden="true" class="w-6 h-6 text-danaherlightblue-500" data-di-res-id="f8b21aa6-f30b99b3" data-di-rand="1708403729282">
+            <path fill-rule="evenodd" d="M2.25 12c0-5.385 4.365-9.75 9.75-9.75s9.75 4.365 9.75 9.75-4.365 9.75-9.75 9.75S2.25 17.385 2.25 12zm13.36-1.814a.75.75 0 10-1.22-.872l-3.236 4.53L9.53 12.22a.75.75 0 00-1.06 1.06l2.25 2.25a.75.75 0 001.14-.094l3.75-5.25z" clip-rule="evenodd"></path>
+        </svg>`;
```
Comment (mhaack) [inline blocks/timeline/timeline.js:20]: Lets move this to an SVG file.

File: blocks/timeline/timeline.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+function updateTimeLine(target) {
+  const clickedMenu = target.closest('.menu-item');
+  if (!clickedMenu) return;
+
+  const main = document.querySelector('main');
+  const timelines = main.querySelectorAll('div.timeline:not(.menu) > div');
+  const nonAutomation = [...timelines].filter((timeline) => !timeline.querySelector('div > p > em').textContent.includes('LABORATORY AUTOMATION'));
+  const nonAnalytical = [...timelines].filter((timeline) => !timeline.querySelector('div > p > em').textContent.includes('ANALYTICAL TOOLS'));
+  const nonDigital = [...timelines].filter((timeline) => !timeline.querySelector('div > p > em').textContent.includes('DIGITAL SOLUTIONS'));
```
Comment (mhaack) [inline blocks/timeline/timeline.js:42]: This looks like it is easy to break? What about other terms, lower cases etc.

File: blocks/timeline/timeline.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      break;
+    case 'Analytical Tools':
+      updateEachTimeline(nonAnalytical, timelines);
+      break;
+    case 'Digital Solutions':
+      updateEachTimeline(nonDigital, timelines);
+      break;
+    default:
+      updateEachTimeline([], timelines);
+      break;
```
Comment (mhaack) [inline blocks/timeline/timeline.js:58]: This looks like it is easy to break? What about other terms, lower cases etc.

File: blocks/timeline/timeline.js (hunk 1 above)
Comment (rgravitvl) [reply blocks/timeline/timeline.js]: Done

File: blocks/timeline/timeline.js (hunk 3 above)
Comment (rgravitvl) [reply blocks/timeline/timeline.js]: Used id instead of text content. Done

File: blocks/timeline/timeline.js (hunk 2 above)
Comment (rgravitvl) [reply blocks/timeline/timeline.js:41]: I see, this the only option to filter and current implementation is also use the same. Thanks.

**Review by mhaack — APPROVED** [review]

**Pull request merged**